src/foam/u2/view/CollapseableDetailView.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'CollapseableDetailView',
+  extends: 'foam.u2.View',
+  documentation: `
+    A detail view that initially just renders the summary of the object
+    with a "Show" button that, when clicked, shows a detail view of that
+    object.
+  `,
+  requires: [
+    'foam.u2.layout.Cols',
+  ],
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'isCollapsed',
+      value: true
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'view',
+      value: { class: 'foam.u2.DetailView' }
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'citationView',
+      expression: function(data) {
+        return {
+          class: 'foam.u2.CitationView',
+          of: data && data.cls_
```

Comment:
Does citation view need an of? It doesn't appear to use it.

Reply:
My understanding is that it's needed for the facet to work. Am I wrong on that?